tomviz/ImageStackDialog.cxx

```diff
@@ -18,20 +18,296 @@
+}
+
+void ImageStackDialog::onOpenFolderClick()
+{
+  openFileDialog(QFileDialog::Directory);
+}
+
+void ImageStackDialog::openFileDialog(int mode)
+{
+  QStringList filters;
+  filters << "TIFF Image files (*.tiff *.tif)";
+
+  QFileDialog dialog(nullptr);
+  if (mode == QFileDialog::ExistingFiles) {
+    dialog.setFileMode(QFileDialog::ExistingFiles);
+    dialog.setNameFilters(filters);
+  } else if (mode == QFileDialog::Directory) {
+    dialog.setFileMode(QFileDialog::Directory);
+    dialog.setOption(QFileDialog::ShowDirsOnly, true);
+  }
+
+  if (dialog.exec()) {
+    QStringList fileNames;
+    if (mode == QFileDialog::ExistingFiles) {
+      processFiles(dialog.selectedFiles());
+    } else if (mode == QFileDialog::Directory) {
+      processDirectory(dialog.selectedFiles()[0]);
+    }
+  }
+}
+
+void ImageStackDialog::processDirectory(QString path)
+{
+  QStringList fileNames;
+  QDir directory(path);
+  foreach (auto file, directory.entryList(QDir::Files)) {
+    fileNames << directory.absolutePath() + QDir::separator() + file;
+  }
+  processFiles(fileNames);
+}
+
+void ImageStackDialog::processFiles(QStringList fileNames)
+{
+  QStringList fNames;
+  foreach (auto file, fileNames) {
+    if (file.endsWith(".tif") || file.endsWith(".tiff")) {
+      fNames << file;
+    }
+  }
+  QList<ImageInfo> summary = LoadStackReaction::loadTiffStack(fNames);
+
+  bool isVolume = false;
+  bool isTilt = false;
+  bool isNumbered = false;
+  DataSource::DataSourceType stackType = DataSource::DataSourceType::Volume;
+
+  isVolume = detectVolume(fNames, summary);
+  if (!isVolume) {
+    isTilt = detectTilt(fNames, summary);
+    if (isTilt) {
+      stackType = DataSource::DataSourceType::TiltSeries;
+    } else {
+      isNumbered = detectVolume(fNames, summary, false);
+      if (!isNumbered) {
+        defaultOrder(fNames, summary);
+      }
+    }
+  }
+  this->setStackType(stackType);
+  this->setStackSummary(summary);
+}
+
+bool ImageStackDialog::detectVolume(QStringList fileNames,
+                                    QList<ImageInfo>& summary, bool matchPrefix)
+{
+  if (fileNames.size() < 1) {
+    return false;
+  }
+  QString thePrefix;
+  QString prefix;
+  int num;
+  QRegExp volRegExp("^(.*[^\\d]{1})(\\d+)(\\.(tif|tiff))$");
+
+  for (int i = 0; i < fileNames.size(); ++i) {
+    if (volRegExp.exactMatch(fileNames[i])) {
+      prefix = volRegExp.cap(1);
+      if (i == 0) {
+        thePrefix = prefix;
+      }
+      if (prefix == thePrefix || !matchPrefix) {
+        num = volRegExp.cap(2).toInt();
+        summary[i].pos = num;
+      } else {
+        return false;
+      }
+    } else {
+      return false;
+    }
+  }
+  return true;
+}
+
+bool ImageStackDialog::detectTilt(QStringList fileNames,
+                                  QList<ImageInfo>& summary, bool matchPrefix)
+{
+  if (fileNames.size() < 1) {
+    return false;
+  }
+  QString thePrefix;
+  QString prefix;
+  int num;
+  QString sign;
+  QString num_;
```

Comment:
Rather than using num_ it will probably be easier to read if you call it numStr, or differentiate num and num_ so it is clear what each is for.